nova/conductor/tasks/live_migrate.py

```diff
@@ -106,13 +108,54 @@ def _check_destination_is_not_source(self):
             raise exception.UnableToMigrateToSelf(
                     instance_id=self.instance.uuid, host=self.destination)
 
+    def _get_aggregate_metadata(self,instance_uuid, dest):
+        aggregate_list = db.aggregate_get_by_host(self.context, self.destination)
+        LOG.debug(_("lwatta::====> aggregate_list size %s"), len(aggregate_list))
+        if len(aggregate_list) == 1:
+            return aggregate_list[0]
+
+        # Check to see if host is in an aggregate or in more than one aggregate (cause we do stupid stuff)
+        if len(aggregate_list) > 1:
+            agg_name = []
+            for i in aggregate_list:
+                agg_name.append(i.name)
+                LOG.debug(_("lwatta::====> agg_name %s"), agg_name)
+                reason = _("Unable to migrate %(instance_uuid)s to %(dest)s: "
+                "Destination host is in more then one aggregate:%(agg_name)s")
+                raise exception.MigrationPreCheckError(reason=reason % dict(
+                    instance_uuid=instance_uuid, dest=dest, agg_name=agg_name))
+        elif len(aggregate_list) < 1:
+            LOG.debug(_("lwatta::====> Host is not in an aggregate %s"), dest)
+            reason = _("Unable to migrate %(instance_uuid)s to %(dest)s: "
+            "Destination host is not in any aggregates")
+            raise exception.MigrationPreCheckError(
+                reason=reason % dict(instance_uuid=instance_uuid, dest=dest))
+
     def _check_destination_has_enough_memory(self):
-        avail = self._get_compute_info(self.destination)['free_ram_mb']
+        instance_uuid = self.instance.uuid
+        dest = self.destination
+        aggr = self._get_aggregate_metadata(instance_uuid, dest)
+        # for icehouse replace aggr.metadetails with aggr.metadata.
+
+        # (schoksey replaced) ram_ratio = float(aggr.metadetails['ram_allocation_ratio'])
+        ram_ratio = float(aggr.get('metadetails')['ram_allocation_ratio'])
```

Comment:
I will test without ram_allocation set tonight. Most likely i will need to add a quick check. If there is no ram_allocation use default of 1.5